bundles/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/internal/handler/FeedHandler.java

```diff
@@ -113,13 +128,91 @@ private void refreshFeedState() {
         boolean feedUpdated = updateFeedIfChanged(feed);
 
         if (feedUpdated) {
+            publishDynamicChannelsIfLinked();
+
             List<Channel> channels = getThing().getChannels();
             for (Channel channel : channels) {
                 publishChannelIfLinked(channel.getUID());
             }
         }
     }
 
+    private void publishDynamicChannelsIfLinked() {
+        List<SyndEntry> entries = currentFeedState.getEntries()
+                .stream()
+                .limit(numberOfEntries.intValue())
+                .collect(Collectors.toList());
+
+        createDynamicChannels(numberOfEntries.intValue());
+        setUnusedDynamicChannelsToUndef(entries.size());
+
+        for (int i = 0; i < entries.size(); i++) {
+            SyndEntry entry = entries.get(i);
+
+            ChannelUID titleChannelUID = new ChannelUID(titleChannelGroupUID, String.valueOf(i));
+            if (isLinked(titleChannelUID)) {
```

Comment:
Since you already have the value of the entry (and that seems to be the expensive operation), do you really need to check `isLinked`?